File: output/ChatGPT/codenet/Java/C++/s393796709.cpp
#include <iostream>
#include <vector>
using namespace std;

const long long MOD = 1000000007;

void solve() {
    vector<int> xx(2);
    string line;
    getline(cin, line);
    size_t pos = 0;
    for (int i = 0; i < 2; i++) {
        xx[i] = stoi(line.substr(pos), &pos);
    }
    int min = min(xx[0], xx[1]);
    int max = max(xx[0], xx[1]);
    long long res;
    if (min == max) {
        long long f = 1;
        for (int i = 2; i <= min; i++) {
            f = (f * i) % MOD;
        }
        res = (2 * f * f) % MOD;
    } else if (min + 1 == max) {
        long long f = 1;
        for (int i = 2; i <= min; i++) {
            f = (f * i) % MOD;
        }
        res = (f * ((f * max) % MOD)) % MOD;
    } else {
        res = 0;
    }
    cout << res << endl;
}

int main() {
    solve();
    return 0;
}

File: output/ChatGPT/codenet/Java/Python/s393796709.py
MOD = 1000000007

def solve():
    xx = list(map(int, input().split()))
    min_val = min(xx)
    max_val = max(xx)
    res = 0
    if min_val == max_val:
        f = factorial(min_val)
        res = (2*f*f) % MOD
    elif min_val + 1 == max_val:
        f = factorial(min_val)
        res = (f * ((f*max_val) % MOD)) % MOD
    
    print(res)

def factorial(n):
    res = 1
    for i in range(2, n+1):
        res = (res * i) % MOD
    return res

if __name__ == "__main__":
    solve()